internal/repository/hook.go

```diff
@@ -35,11 +36,13 @@ func (r *Repository) UpdateHook(hookType HookType, content []byte, force bool) e
 	if tree == nil {
 		return fmt.Errorf("worktree is nil, can't update hooks")
 	}
+	slog.Debug("Fetched repository's worktree")
```

Comment:
I don't know, I'm struggling to judge what information is helpful and isn't, I might be too close to the implementation? I'd like to reduce noise as much as possible so the log is valuable, and I'll admit I don't have a lot of experience with threading this needle. @wlynch could you take a look at this PR too?